lib/components/diode.ts

```diff
@@ -1,6 +1,20 @@
 import { commonComponentProps, lrPolarPins } from "lib/common/layout"
-import type { z } from "zod"
+import { z } from "zod"
+import type { Connections } from "./chip"
+
+const connectionTarget = z
+  .string()
+  .or(z.array(z.string()).readonly())
+  .or(z.array(z.string()))
+
+const connectionsProp = z
+  .custom<Connections>()
+  .pipe(z.record(z.string(), connectionTarget))
+
+export const diodeProps = commonComponentProps.extend({
+  connections: connectionsProp.optional(),
```

Comment:
Lets make it so that a diodes connection keys are fixed to anode, cathode, pin1, pin2, pos and neg